listenbrainz/mbid_mapping/mapping/mb_metadata_cache.py

```diff
@@ -414,6 +431,67 @@ def create_cache(mb_conn, mb_curs, lb_conn, lb_curs):
     log("mb metadata cache: done")
 
 
+def delete_rows(lb_curs, recording_mbids):
+    query = """
+        DELETE FROM mb_metadata_cache
+              WHERE recording_mbid IN %s
+    """
+    lb_curs.execute(query, (tuple(recording_mbids),))
+
+
+def config_postgres_join_limit(mb_curs):
+    # Because of the size of query we need to hint to postgres that it should continue to use joins
+    mb_curs.execute('SET geqo = off')
+    mb_curs.execute('SET geqo_threshold = 20')
+    mb_curs.execute('SET from_collapse_limit = 15')
+    mb_curs.execute('SET join_collapse_limit = 15')
+
+
+def update_dirty_cache_items(mb_curs, lb_conn, lb_curs):
+    dirty_query = """
+        SELECT recording_mbid
+          FROM mb_metadata_cache
+         WHERE dirty = 't'
+    """
+
+    log("mb metadata update: getting dirty items")
+    lb_curs.execute(dirty_query)
+    recording_mbids = lb_curs.fetchall()
+
+    config_postgres_join_limit(mb_curs)
+
+    log(f"mb metadata update: got {len(recording_mbids)} to update")
+
+    log("mb metadata update: Running looooong query on dirty items")
+    query = get_metadata_cache_query(with_values=True)
+    values = [(row[0],) for row in recording_mbids]
+    psycopg2.extras.execute_values(mb_curs, query, values, page_size=len(values))
```

Comment:
How many rows do you expect to update in one run of this job? Setting page size to insert all rows in 1 insert may be problematic, I vaguely remember issues in inserting LB stats or YIM when stats payload were huge. Though here the JSON will be much smaller so probably not an issue. In any case, having an estimate as to what we are expecting here would be nice.

Reply:
Uh, I see. This cursor is iterated for results afterwards so page_size has to be eqaul to len otherwise won't work.